components/infra-proxy-service/server/nodes.go

```diff
@@ -201,3 +263,19 @@ func safeStringFromMapFloat(values map[string]interface{}, key string) string {
 	}
 	return strconv.FormatFloat(values[key].(float64), 'E', -1, 64)
 }
+
+// This returns the value referenced by `key` in `values`. If value is nil,
+// it returns an empty slice string; otherwise it returns the original slice string.
```

Comment:
Please add unit tests. (From your comment I am not even quite sure if there are two or three cases here.)

Reply:
@msorens moved some of the helpers into `util.go` and added test cases into file `util_test.go`

Could you please have a look and let us know further? Thanks!